src/sir_kibble.c

```diff
@@ -0,0 +1,309 @@
+    if (arg0->base.xspeed < 0) {
+        arg0->base.xspeed += 0xe;
+        if (arg0->base.xspeed >= 1) {
+            arg0->base.xspeed = 0;
+        }
+    }
+    else {
+        arg0->base.xspeed -= 0xe;
+        if (arg0->base.xspeed <= -1) {
+            arg0->base.xspeed = 0;
+        }
+    }
+
+    if ((arg0->base.unk62 & 4) == 0) {
+        arg0->base.yspeed -= 0x2b;
+        if (arg0->base.yspeed < -0x300) {
+            arg0->base.yspeed = -0x300;
+        }
+    }
+
+    if (arg0->base.counter == 0x78) {
+        arg0->base.yspeed = 0x400;
+        PlaySfx(&arg0->base, 0x145);
+    }
+
+    if (arg0->base.counter >= 0x79) {
+        if (arg0->base.unk62 & 4) {
+            ObjectSetFunc(arg0, 0, sub_080B1368);
+            if (arg0->object->subtype1) {
+                arg0->unk83 = 1;
+                arg0->base.xspeed = 0x80;
+                if (arg0->base.flags & 1) {
+                    arg0->base.xspeed = -arg0->base.xspeed;
+                }
+            }
+            return;
+        }
+
+    }
+    if (arg0->base.counter == 0x32) {
+        sub_080B1DF8(arg0);
+    }
+    ++arg0->base.counter;
+}
+
+void sub_080B16A8(struct ObjectBase *arg0) {
+    arg0->flags |= 4;
+    if (arg0->xspeed < 0) {
+        arg0->xspeed += 0xe;
+        if (arg0->xspeed >= 1) {
+            arg0->xspeed = 0;
+        }
+    }
+    else {
+        arg0->xspeed -= 0xe;
+        if (arg0->xspeed <= -1) {
+            arg0->xspeed = 0;
+        }
+    }
+    arg0->counter++;
+}
+
+void* CreateUnknownA0(struct Object* arg0, u8 arg1) {
+    struct Object2 *obj, *obj2;
+    struct Task* task = TaskCreate(ObjectMain, sizeof(struct Object2), 0x1000, TASK_USE_EWRAM, ObjectDestroy);
+    obj2 = TaskGetStructPtr(task);
+    obj = obj2;
+    InitObject(obj, arg0, arg1);
+    obj->base.flags |= 0x02000140;
+    obj->base.unk5C |= 0x20;
+    sub_0803E2B0(&obj->base, -5, -3, 5, 8);
+    sub_0803E308(&obj->base, -6, -4, 6, 10);
+    ObjectInitSprite(obj);
+    obj->base.sprite.unk14 = 0x640;
+    if (arg0->subtype1 & 1) {
+        obj->base.flags |= 1;
+    }
+    gUnk_08351648[obj->type].unk10(obj);
+
+    return obj;
+}
+
+void sub_080B17BC(struct ObjectBase *arg0) {
+    if (arg0->parent) {
+        struct ObjectBase *parent = (struct ObjectBase*)arg0->parent;
+        if (parent->flags & 0x1000) {
+            arg0->parent = 0;
+        }
+        if (sub_0803925C(arg0, parent)) {
+            arg0->flags |= 0x1000;
+        }
+    }
+    if (arg0->flags & 1) {
+        arg0->xspeed += 0x10;
+        if (arg0->xspeed < -0x02a8) {
+            arg0->xspeed = -0x02a8;
+        }
+        else if (arg0->xspeed > 0x02a8) {
+            arg0->xspeed = 0x02a8;
+
+        }
+    }
+    else {
+        arg0->xspeed -= 0x10;
+        if (arg0->xspeed <= 0x02a8) {
+            if (arg0->xspeed < -0x02a8) {
+                arg0->xspeed = -0x02a8;
+            }
+
+        }
+        else {
+            arg0->xspeed = 0x02a8;
+        }
```

Comment:
I wonder if it matches if you follow the if - else if statement above.

Reply:
Done! My bad :sweat_smile: